blog/2023-06-16-inline-sdk.mdx

```diff
@@ -0,0 +1,1104 @@
+    ...
+    // highlight-start
+    return {
+        playerState,
+        togglePlay,
+        handleOnTimeUpdate,
+        handleVideoSpeed,
+        toggleMute,
+        seekTo
+    }
+    //highlight-end
+}
+```
+
+With everything put together, your `useVideoPlayer` hook should look like this:
+
+```ts title="src/bands/VideoBand/useVideoPlayer.ts"
+import { useState, useEffect } from "react";
+
+const useVideoPlayer = (
+    videoElement: React.MutableRefObject<HTMLVideoElement>
+) => {
+    const [playerState, setPlayerState] = useState({
+        isPlaying: false,
+        progress: 0,
+        speed: 1,
+        isMuted: false,
+    });
+
+    const togglePlay = () => {
+        setPlayerState({
+            ...playerState,
+            isPlaying: !playerState.isPlaying,
+        });
+    };
+
+    const handleOnTimeUpdate = () => {
+        const progress =
+            (videoElement.current.currentTime /
+                videoElement.current.duration) *
+            100;
+        setPlayerState({
+            ...playerState,
+            progress,
+        });
+    };
+
+    const seekTo = (value: number) => {
+        if (value >= 0 && value <= 100) {
+            videoElement.current.currentTime =
+                (videoElement.current.duration / 100) * value;
+            setPlayerState({
+                ...playerState,
+                progress: value,
+            });
+        }
+    };
+
+    const handleVideoSpeed = (event) => {
+        const speed = +event.target.value;
+        videoElement.current.playbackRate = speed;
+        setPlayerState({
+            ...playerState,
+            speed,
+        });
+    };
+
+    const toggleMute = () => {
+        setPlayerState({
+            ...playerState,
+            isMuted: !playerState.isMuted,
+        });
+    };
+
+    useEffect(() => {
+        videoElement.current.muted = playerState.isMuted;
+    }, [playerState.isMuted, videoElement]);
+
+    useEffect(() => {
+        videoElement.current.playbackRate = playerState.speed;
+    }, [playerState.speed, videoElement]);
+
+    useEffect(() => {
+        playerState.isPlaying
+            ? void videoElement.current.play()
+            : void videoElement.current.pause();
+    }, [playerState.isPlaying, videoElement]);
+
+    return {
+        playerState,
+        togglePlay,
+        handleOnTimeUpdate,
+        handleVideoSpeed,
+        toggleMute,
+        seekTo,
+    };
+};
+
+export default useVideoPlayer;
+```
+
+### Creating the component
+
+Now, add the hook to your custom band. Open `src/bands/VideoBand/Band.tsx`. Here you will find a simple react component. This component is the entrypoint for the band. In the `Band` component, add a `player` ref and add your `useVideoPlayer` hook. Add the `props` function parameter to the component.
+
+```tsx title="src/bands/VideoBand/Band.tsx"
+import React, { useRef } from "react";
+import { BandProps } from "inline";
+import "./Band.css";
```

Comment:
I would take this line out and add it in its own step after you author the `Band.css`.